addons/microsoft_calendar/static/src/views/microsoft_calendar/microsoft_calendar_controller.js

```diff
@@ -19,17 +19,15 @@ patch(AttendeeCalendarController.prototype, "microsoft_calendar_microsoft_calend
             [[this.user.userId]],
         );
         const syncResult = await this.model.syncMicrosoftCalendar();
-        if (["need_auth", "need_config_from_admin"].includes(syncResult.status)) {
+        if (syncResult.status === "need_auth") {
+            window.location.assign(syncResult.url);
```

Comment:
@thiryr 

Just to make sure.

Seems like pre-owl code also called a "configureCalendarProviderSync" when the response was "need_auth".
And the content of that method seems similar to what we have now in "configureCalendarProviderSync", that is still called, in a different flow.

![image](https://user-images.githubusercontent.com/41282536/214015785-dc6ca10a-ce40-4c80-ac28-881f5adafbaf.png)

And now we redirect to an URL.

Did the flow change between that version and now?
I don't know calendar sync very well so it could be normal.

cc @arj-odoo

Reply:
@awa-odoo Ah you're right. I thought that was overriden.

The change actually happens a bit earlier here:

https://github.com/odoo-dev/odoo/commit/ba2b0020dfc2a6484fd4a9f2ed72eece414a823a#diff-9ee47a179b53ecf8f646f6e7e672adc3720cebfcb17b9a166846554850e7f087L114

This is when we added the configuration modal (I think you worked a bit on this with flch?)

Reply:
(confirmed with flch that this wasn't intended)

Reply:
Another blip!
Actually, it seems like he did that because of the issue where if the module is installed, but not configured. You get need_auth, and the user is redirected to nowhere-land.

Maybe that can be another PR on this task, so we don't leave this issue hanging while I figure out how to do it? @awa-odoo

Reply:
@thiryr That depends a bit.

If you tell me that the issue is completely different from this or that it requires a long time to figure out / fix, then yes we can make a separate PR.
Otherwise, it's always better to fix the whole flow in one singular fix, to avoid having back-and-forth changes.

Reply:
Indeed, maybe I can come up with a quick patch for the rest of the flow so probably not worth a new PR.
----------

I think I can do something fairly quickly to just check that the URL will at least be OK

But there is no programatical way of testing the client ID and Secret for OAuth with Google. They just don't provide the infrastructure (you're expected to generate a key manually to test your app, which goes against the purpose of doing this in the first place).

I'll see if we can be more explicit about the key being wrong, in case an OAuth fails because of that once a user actually tries connecting.

Reply:
Ideally the controller should provide a separate key and not "need auth" with a wrong URL.
Or at least blank the URL so we can test it here.

I'm not sure I get all the implications though, maybe something to check with Florian directly.
If it takes bigger proportions, we may re-evaluate and create a separate PR.

Thank you for investigating :slightly_smiling_face:

Reply:
Ok, so I managed to get *something* going for Google (need to check the actual url path, not great). 

But for Azure, I can't even rely on an error page (it requires that you try to connect before getting an error). I would need to ask the user for more information and try generating a bearer token for some Azure service. It seems like all these services require the tenant ID (which is not required for OAuth).

This doesn't seem worth it just to check for a typo, or user misconfiguration, knowing that users would quickly notice it. 99% of the time, the admin would try it right away and notice it doesn't work. For Google they wouldn't even need to try connecting, if they configure it through the modal they'll get an error screen straight away.

Checked with lna, he said it's fine to just let it be invalid. As we apparently do that for social (e.g. facebook).
Let me know if you think of some case where this is an issue.

Reply:
Well I believed we checked everything we had too on this one ^^ Let's go then!

Thanks for the investigation!